core/mavlink_system.cpp

```diff
@@ -544,92 +623,123 @@ void Device::set_flight_mode_async(FlightMode device_mode, command_result_callba
+    _params.get_param_async(name, std::bind(&MAVLinkSystem::receive_int_param, _1, _2,
                                             callback));
 }
 
-void Device::get_param_ext_float_async(const std::string &name,
-                                       get_param_float_callback_t callback)
+void MAVLinkSystem::get_param_ext_float_async(const std::string &name,
+                                              get_param_float_callback_t callback)
 {
-    _params.get_param_async(name, std::bind(&Device::receive_float_param, _1, _2,
+    _params.get_param_async(name, std::bind(&MAVLinkSystem::receive_float_param, _1, _2,
                                             callback), true);
 }
 
-void Device::get_param_ext_int_async(const std::string &name,
-                                     get_param_int_callback_t callback)
+void MAVLinkSystem::get_param_ext_int_async(const std::string &name,
+                                            get_param_int_callback_t callback)
 {
-    _params.get_param_async(name, std::bind(&Device::receive_int_param, _1, _2,
+    _params.get_param_async(name, std::bind(&MAVLinkSystem::receive_int_param, _1, _2,
                                             callback), true);
 }
 
-void Device::receive_float_param(bool success, MavlinkParameters::ParamValue value,
-                                 get_param_float_callback_t callback)
+void MAVLinkSystem::receive_float_param(bool success, MAVLinkParameters::ParamValue value,
+                                        get_param_float_callback_t callback)
 {
     if (callback) {
         callback(success, value.get_float());
     }
 }
 
-void Device::receive_int_param(bool success, MavlinkParameters::ParamValue value,
-                               get_param_int_callback_t callback)
+void MAVLinkSystem::receive_int_param(bool success, MAVLinkParameters::ParamValue value,
+                                      get_param_int_callback_t callback)
 {
     if (callback) {
         callback(success, value.get_int());
     }
 }
 
-MavlinkCommands::Result Device::send_command_with_ack(
-    uint16_t command, const MavlinkCommands::Params &params, uint8_t component_id)
+uint8_t MAVLinkSystem::get_autopilot_id() const
+{
+    for (auto compid : _components)
+        if (compid == MAVLinkCommands::DEFAULT_COMPONENT_ID_AUTOPILOT) {
+            return compid;
+        }
+    // FIXME: Not sure what should be returned if autopilot is not found
+    return uint8_t(0);
+}
+
+std::vector<uint8_t> MAVLinkSystem::get_camera_ids() const
+{
+    std::vector<uint8_t> camera_ids;
+    camera_ids.clear();
+
+    for (auto compid : _components)
+        if (compid >= MAV_COMP_ID_CAMERA && compid <= MAV_COMP_ID_CAMERA6) {
+            camera_ids.push_back(compid);
+        }
+    return camera_ids;
+}
+
+uint8_t MAVLinkSystem::get_gimbal_id() const
+{
+    for (auto compid : _components)
+        if (compid == MAV_COMP_ID_GIMBAL) {
+            return compid;
```

Comment:
Same error here, I think.

Reply:
Here, I return gimbal `compid` if I find it.